static/js/recent_topics.js

```diff
@@ -116,9 +116,18 @@ function set_table_focus(row, col) {
     }
 
     // Setting focus after the render is complete doesn't partially hide the row from view.
-    setTimeout(() => {
-        topic_rows.eq(row).find(".recent_topics_focusable").eq(col).children().trigger("focus");
-    }, 0);
+    const topic_row = topic_rows.eq(row);
+    topic_row.find(".recent_topics_focusable").eq(col).children().trigger("focus");
+
+    // Scroll the topic row in the middle
+    // NOTE: Although, according to
+    // https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView#browser_compatibility
+    // `scrollIntoView` is not fully supported on Safari,
+    // it works as intended on Safari v14.0.3 on macOS Big Sur.
+    topic_row.get()[0].scrollIntoView({
+        block: "center",
+    });
```

Comment:
I wonder if we want to try to borrow more of the message_viewport scrolling logic.

Reply:
`message_viewport` scrolling logic is taking care of a lot of things. I think we can get away with keeping it simpler here.

I pushed after fixing a bug that we were not rendering more rows until the user was at the last row of in the recent topics table which we don't want now.